packages/proxy/lib/http/util/cookies.ts

```diff
@@ -3,13 +3,49 @@ import type Debug from 'debug'
 import { URL } from 'url'
 import { cors } from '@packages/network'
 import { AutomationCookie, Cookie, CookieJar, toughCookieToAutomationCookie } from '@packages/server/lib/util/cookies'
+import { calculateSiteContext } from './top-simulation'
+import type { RequestCredentialLevel, RequestResourceType } from '../../types'
 
 interface RequestDetails {
   url: string
   isAUTFrame: boolean
   needsCrossOriginHandling: boolean
 }
 
+export const shouldAttachAndSetCookies = (requestUrl: string, AUTUrl: string | undefined, resourceType?: RequestResourceType, credentialLevel?: RequestCredentialLevel): boolean => {
+  if (!AUTUrl) return false
+
+  const siteContext = calculateSiteContext(requestUrl, AUTUrl)
```

Comment:
You are only using siteContext === 'same-origin'? Do we need to calculate the site context or can we just ensure it's the same origin?

Reply:
In this case, we just need to make sure the request is `same-origin` or `cross-origin`, which I figure is easier to read with `calculateSiteContext` instead of `urlOriginsMatch. If a request is cross origin then we need to consider credential level (unless it's `fetch` with `omit`). 

It looks a bit weird since `calculateSiteContext` is only used in the `shouldAttachAndSetCookies` here, but is leverage in the cookie jar in the [next PR](https://github.com/cypress-io/cypress/pull/23888/files#diff-8bcc53731d8dc8219a282445e0ae5ae47e2884d0fb28ead3eef2b488819d1bf0R137)